app/api_client/base.py

```diff
@@ -34,11 +36,25 @@ def pretty_print_request(prep):
         ))
 
 
+class GenericRequester(object):
+    def __init__(self, client):
+        self.client = client
+
+    def __getattr__(self, name):
```

Comment:
This is a little bit of trickery that takes a method name: eg api_client.req.get_domains, and generates and api call from the method name used.

get_domains -> get | domains
get -> GET requests
domains -> /domains path 

So the api client makes a get request to domains (the rest of the parameters are passed as normal).

If we expanded the usage of this we could use one method like this for every API call,

get_domains
post_suppliers
patch_address

etc etc, eliminating the need for an API client entirely. (!)